test/uploader_spec.js

```diff
@@ -713,24 +713,33 @@ describe("uploader", function () {
       });
     });
   });
-  it("should support unsigned uploading using presets", function () {
+  it("should support unsigned uploading using presets", async function () {
     this.timeout(helper.TIMEOUT_LONG);
-    let presetName;
-    return cloudinary.v2.api.create_upload_preset({
+
+    let preset = await cloudinary.v2.api.create_upload_preset({
       folder: "upload_folder",
       unsigned: true,
       tags: UPLOAD_TAGS,
-    }).then(function (preset) {
-      presetName = preset.name;
-      return cloudinary.v2.uploader.unsigned_upload(IMAGE_FILE, preset.name, {
-        tags: UPLOAD_TAGS,
-      });
-    }).then(function ({ public_id }) {
-      expect(public_id).to.match(/^upload_folder\/[a-z0-9]+$/);
-    }).finally(function () {
-      return cloudinary.v2.api.delete_upload_preset(presetName);
+    }).catch((err) => {
+      console.log(err);
+      throw new Error('create_upload_preset failed');
+    });
+
+    let uploadResponse = await cloudinary.v2.uploader.unsigned_upload(IMAGE_FILE, preset.name, {
+      tags: UPLOAD_TAGS,
+    }).catch((err) => {
+      console.log(err);
+      throw new Error('unsigned_upload failed');
+    });
+
+    expect(uploadResponse.public_id).to.match(/^upload_folder\/[a-z0-9]+$/);
```

Comment:
@patrick-tolosa , isn't that redundant if the upload response failed?

Reply:
Well , yes. It's also redundant if the upload_preset failed, however, I prefer to keep `expects` outside of conditions (if, try) to make it clear what the test is testing. (Same reason I try to keep `return` statements outside of conditions)

What do you think?

Reply:
@patrick-tolosa , I understand the approach and agree it's less readable when the asserts happens inside the `then` clause. 
Since the test will fail anyway when the error is thrown, I'm ok with leaving as-is